controllers/complianceeventsapi/server.go

```diff
@@ -1051,15 +1050,37 @@ func postComplianceEvent(db *sql.DB, cfg *rest.Config, w http.ResponseWriter, r
 
 	reqEvent.Event.PolicyID = policyFK
 
-	err = reqEvent.Create(r.Context(), db)
+	err = reqEvent.Create(r.Context(), serverContext.DB)
 	if err != nil {
 		if errors.Is(err, errDuplicateComplianceEvent) {
 			writeErrMsgJSON(w, "The compliance event already exists", http.StatusConflict)
 
 			return
 		}
 
-		log.Error(err, "error inserting compliance event")
+		pqErr, ok := err.(*pq.Error) //nolint:errorlint
+		if ok && pqErr.Code == "23503" {
```

Comment:
Would it be helpful to make this magic number a const? It seems something like `postgresForeignKeyViolationCode` might be descriptive.